### ISSUE TITLE
add a 'start in dev mode' command for maven and npm

### ISSUE THREAD

**Issue by vlukashov** [vaadin/base-starter-connect#6]
When evaluating the Vaadin Connect base starter...
 - ... as a frontend developer I want to start the project with `npm start` so that I do not have to read any docs and can apply the default approach common for frontend projects.
 - ... as a Java developer I want to start the project with `./mvnw` so that I do not have to read any docs and can apply the default approach common for Java / Maven projects.

DoD:
 - running `npm start` or `./mvnw` when the backend is not started yet:
   - starts the backend (in the dev mode)
   - starts the frontend (in the dev mode)
   - starts a watcher that regenerates the OpenAPI spec file after every relevant source file change
   - starts a watcher that regenerates the default JS client config and the JS service wrappers after every relevant change to the properties file / the OpenAPI spec file
 - the README is updated with the accurate 'how to start' information

**Comment by SomeoneToIgnore** [comment]
I think we should define the way our framework will be used by the developers before we start implementing this and similar tickets.

The way I see it now is that the Java part is a completely standalone product — it can exist without the frontend part, accepting POST requests, that can be done via spring tools in backend IT tests.
For debugging the backend part, a backend developer needs to start a "special" Java process with special parameters, which is usually done via IDE, but, again, no frontend part is involved into the process and it seems like it should be this way.

Also, since we assume that both `maven` and `node` are installed on the developer's machine, running a frontend part is a matter of executing one standalone console command (`npm run start:frontend` currently) so it might be fine to let it be this way as long as we document it.
In addition, the backend developer should be able to build a production bundle of the frontend part and run both client and backend part with a single maven command.

As for the real application IT tests, currently they are considered to be a part of a frontend project tests, not a backend one.
There we are already able to launch both backend and frontend part, so we're fine.

**Comment by SomeoneToIgnore** [comment]
In addition to the things mentioned above, there are also potential complications from the technical point of view: it's relatively complicated to start both backend and frontend servers from Maven.

The main pain point here is the we're not able to chain plugin goals, meaning that when someone runs `mvn spring-boot:run`, we cannot detect that and trigger any other plugin execution there.
Neither we can create a custom goal that calls `mvn spring-boot:run` before or after it.

What we can do in the pom is to add a profile that runs a custom task when the profile is enabled, but this is merely the same as running a separate `npm run start:frontend`.
This will increase the pom size and adds another issue: we will need to run `npm install` before, presumably every time before we start the frontend part from Maven, which will increase the start time (adds ~6.5s on my machine for the starter and all packages already installed).

Another concern would be the fact that it's hard to add more feature, if they are required later, in a pom.xml — sooner or later we'll be forced to create a plugin that does this programmatically in Java.
But this will be duplicating the node scripts we already have and seems redundant, since the CLI is planned for the future.

**Comment by SomeoneToIgnore** [comment]
TL;DR

It's either time-consuming or bad DX solution that we can make currently to fix the ticket and it's not clear why do we need to to this at all, so I propose to postpone the ticket.

**Comment by vlukashov** [comment]
won't implement starting the entire app with `./mvnw`
